addons/crm/models/stage_time_mixin.py

```diff
@@ -0,0 +1,88 @@
+from odoo import fields, models
+
+
+class StageTimeMixin(models.AbstractModel):
+    _name = "stage.time.mixin"
+    _description = "Mixin to compute the time a record spent in each stage"
+
+    time_per_stage = fields.Char(compute="_compute_time_per_stage", help="JSON string that maps stage ids to ms and days")
```

Comment:
At some place, an example final structure of that JSON would be useful. Maybe in the `_compute`'s docstring?